x-pack/plugins/actions/server/builtin_action_types/case/schema.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+export const MappingActionType = schema.oneOf([
+  schema.literal('nothing'),
+  schema.literal('overwrite'),
+  schema.literal('append'),
+]);
+
+export const MapRecordSchema = schema.object({
+  source: schema.string(),
+  target: schema.string(),
+  actionType: MappingActionType,
+});
+
+export const CaseConfigurationSchema = schema.object({
+  mapping: schema.arrayOf(MapRecordSchema),
+});
+
+export const ExternalIncidentServiceConfiguration = {
+  apiUrl: schema.string(),
+  casesConfiguration: CaseConfigurationSchema,
+};
+
+export const ExternalIncidentServiceConfigurationSchema = schema.object(
+  ExternalIncidentServiceConfiguration
+);
+
+export const ExternalIncidentServiceSecretConfiguration = {
+  password: schema.string(),
+  username: schema.string(),
+};
+
+export const ExternalIncidentServiceSecretConfigurationSchema = schema.object(
+  ExternalIncidentServiceSecretConfiguration
+);
+
+export const UserSchema = schema.object({
+  fullName: schema.oneOf([schema.nullable(schema.string()), schema.maybe(schema.string())]),
+  username: schema.oneOf([schema.nullable(schema.string()), schema.maybe(schema.string())]),
+});
+
+const EntityInformation = {
+  createdAt: schema.string(),
+  createdBy: UserSchema,
+  updatedAt: schema.nullable(schema.string()),
+  updatedBy: schema.nullable(UserSchema),
+};
+
+export const EntityInformationSchema = schema.object(EntityInformation);
+
+export const CommentSchema = schema.object({
+  commentId: schema.string(),
+  comment: schema.string(),
+  version: schema.maybe(schema.string()),
```

Comment:
You should be using `schema.nullable()` here instead of `schema.maybe()`.  Otherwise, if the field is set at some point, and then you'd like to "null it out", there's no way to do that, as the updates we make to the SO are partial updates, so the effect would be that the field is unchanged.

There are a couple of other references to `schema.maybe()` in here as well.

I'll note that it's unwieldy to use the TS types generated from `schema.nullable()` to generate "writable" data structures in code, as TS will require the fields to be set to `undefined` or `null` - you can't simply just not set the field.  But I believe that's true for `schema.maybe()` as well, and in general, using the the TS types generated from schema to use with programmatically built data structures is ... can be a bit painful.

Reply:
You are right about `nullable`. I do not know why we started with `undefined`. The UI sent null for non-used values. Thanks a lot for catching this.